syntax = "proto3";

package broker;

service BrokerService {
    rpc HealthCheck (HealthCheckRequest) returns (HealthCheckResponse);
    rpc PromoteToPrimary(PromoteToPrimaryRequest) returns (PromoteToPrimaryResponse);
    rpc ReplicateState(ReplicateStateRequest) returns (ReplicateStateResponse);
}

message HealthCheckRequest {}

message HealthCheckResponse {
    bool status = 1;
    string message = 2;
    string timestamp = 3;
}

message PromoteToPrimaryRequest {}

message PromoteToPrimaryResponse {
    bool success = 1;
}

message ReplicateStateRequest {
    string state = 1;
}

message ReplicateStateResponse {
    bool success = 1;
}